### ISSUE TITLE
🎙️ talk.js - March 2020 (previously Feb)

### ISSUE THREAD

**Issue by ongmin** [SingaporeJS/talk.js#32]
### 📝 Details
- [x] Event: [talk.js](https://www.meetup.com/Singapore-JS/)
- [x] Date: 25 March (Wed) _(New Date)_
- [x] Hosts: @ongmin @joeytwiddle
- [x] Venue: Stripe

### 📅 Agenda

Time   | Agenda
------ | :-----
7.00pm | Doors open - _mingle with guests and speakers over 🍕_
7.30pm | Start of event - _opening segment by host_
7.40pm | Start of scheduled talks
8.40pm | Open announcements - _open segment for anyone to make announcements_
9.00pm | End of event

### :speech_balloon: Talks

- [ ] My Awesome Talk by _You_

### :seedling: Propose a talk this month

Want to speak at this month's talk? Leave a comment below with the following details:

<details>
<summary>Click for details</summary>

> #### Title of your topic
> Insert catchy title here
>
> #### Topic description
> Tell us what you'd like to talk about
>
> #### Presenter(s)
> Include GitHub handles where possible. Add your Twitter handle if you wish
>
> #### Presenter's bio
> A short introduction of yourself
>
> #### Duration
> 5 minutes for presenting an introduction to the topic, then up to 20 minutes for discussion.
>
> #### Deck URL (if any)
> You can add this part in later

</details>

### :loudspeaker: Open announcements
Have something you'd like to tell the audience? We have an open announcement segment for you to announce job opportunities, promote a meet up, upcoming conferences, etc.

---

### :pencil: Organizers' checklist

<details>
<summary>Click for details</summary>

- [x] Check available dates on [EngineersSG](https://engineers.sg/)
- [ ] Confirm date and venue availability
- [ ] Create event on [SingaporeJS Meetup Page](https://www.meetup.com/singapore-js/)
- [ ] Inform @engineersftw for video recordings
- [ ] Announce event on [SingaporeJS Gitter](https://gitter.im/SingaporeJS/home) and other social media platforms
- [ ] Prepare deck for event

\* _checked boxes indicate confirmed/done_

</details>

**Comment by wlwl2** [comment]
**Title of your topic**
An Offline Web Time Converter

**Topic description**
I will go through a demo of a simple time converter which uses the Service Worker API.

**Presenter**
[William](https://wlwl2.com/)

**Presenter's bio**
I develop websites, games, and other applications. Topics that I am particularly interested in at the moment include: web, game and mobile development, and Chinese and Japanese translations.

**Duration**
20min + 5min discussion

**Deck URL**
TBA

**Comment by thchia** [comment]
**Title of your topic**
The Monad in JavaScript 😱

**Topic description**
I attempt to discuss what a monad is, what map, apply and bind do and why these things even exist at all. 

**Presenter**
Thomas Chia
www.github.com/thchia
@th__chia on Twitter

**Presenter's bio**
Frontend Developer. Lover not a hater.
Loves include:
- React
- SingaporeJS
- [SingaporeCSS](https://singaporecss.github.io/)
- [React Knowledgeable](www.reactknowledgeable.org)
- Cats

Hates:
- N/A

**Duration**
Not really sure yet but I’m going to try and keep it under 25min. 

**Deck URL (if any)**
Soon

**Comment by ongmin** [comment]
Thanks for the submissions @wlwl2 and @thchia!

We have one more submission from Ankita from Stripe coming in.

**Comment by ankita-stripe** [comment]
**Title of your topic**
A look under the hood of Stripe.js

**Topic description**
Stripe.js are pre-built UI elements that abstract away complexity of integrating with complex payment methods, easing integration experience for developers. In this talk, I will share about the workings of Stripe.js, and explain the design considerations that were made in creating this library. 

**Presenter(s)**
Ankita Gupta
www.github.com/ankitagupta12
@gazubi on twitter

**Presenter's bio**
I work as a Software Engineer at Stripe, working on building global payments infrastructure. I enjoy working on both frontend and backend technologies, as a a part of my work. My interests involve web development, distributed systems, data pipelines, and infrastructure management.

**Duration**
25min presentation + 5 min discussion

**Deck URL (if any)**
Coming Soon!

**Comment by aaronpowell** [comment]
**Title of your topic**
WebAssembly, your browsers sandbox

**Topic description**
We've been doing web development for 30+ years and in all that time have you ever stopped to think, “This SPA needs more C++”? Well thanks to the power of WebAssembly you can finally bring C, C++, Rust, Go and other high level languages to the browser.

So does this mean that we can replace our JavaScript with these other languages? Probably not, so what is the role that WebAssembly can play in building web applications? And most importantly, what does it look like as a web developer to try and incorporate these platforms that have traditionally been on the server?

For this talk, we'll look at some of the fundamentals of WebAssembly, how to incorporate it into our development process and ultimately what it looks like to build an application that uses a mixed development stack.

**Presenter(s)**
Aaron Powell
github.com/aaronpowell
@slace on Twitter
www.aaron-powell.com

**Presenter's bio**
Aaron is a Developer Advocate at Microsoft. Having spent 15 years doing web development he's seen it all, from browser wars, the rise of AJAX and the fall of 20 JavaScript frameworks (and that was just yesterday!). Always tinkering with something new he explores crazy ideas like writing your own implementation of numbers in .NET, creating IoC in JavaScript or implementing tic-tac-toe using git commits.

**Duration**
Flexible, depending on how much time you want to allow me! 😛 (Can be 30 minutes, can be 45 minutes, can be something else)

**Deck URL**
_Coming soon_

**Comment by ongmin** [comment]
Hello @wlwl2 @thchia @ankita-stripe @aaronpowell,

in light of the 2019-nCoV, we're considering changing the meet-up into a virtual one or delaying the talks until next month when there's more clarity on the situation.

What are your opinions/preferences?
Thanks!

**Comment by aaronpowell** [comment]
Hi @ongmin,

I'm happy to do it as a virtual event, as I wouldn't be able to do it in person the following month as I live in Sydney and am going to be in Singapore that week (but that other event may not go ahead, just waiting on leadership decisions).

My only ask is that if it is a virtual event, could I be the first speaker as I'm 3 hours ahead of Singapore 😄.

**Comment by wlwl2** [comment]
@ongmin Even though I want the talk to take place this month for cases like @aaronpowell 's, I think it would be better if the event was delayed to next month just in case less people turn up because it is virtual, or have trouble connecting. Also, you would have to do a different/special preparation (set up video calls and times) for this event, which might confuse people. I'll prepare my talk either way.

**Comment by thchia** [comment]
@ongmin i will also prepare either way, but prefer the usual format of being there in person. TalkCSS did a Google hangout last week (though I couldn’t go) - perhaps they have some feedback on whether it’s worth it?

**Comment by thorsten-stripe** [comment]
@ongmin while we have both temperature screening and increased cleaning measures in place[0], nonetheless it is probably wise to move to live stream instead. We can host the speakers in our training room and stream via zoom directly from the room, or of course any alternative VC platform (would need separate streaming equipment though).

Let me know what you think.

[0] https://www.moh.gov.sg/news-highlights/details/risk-assessment-raised-to-dorscon-orange

**Comment by aaronpowell** [comment]
I've had to cancel my trip to Singapore so I won't be in town next week. I'm still happy to deliver the talk via Zoom/Google Hangouts/etc. if that's an option.

**Comment by ongmin** [comment]
Hi, in light of the responses, we've decided to postpone the talks to March aside from Aaron's. Just for ease of transferring the talk topics, I've renamed this issue to March.

@aaronpowell, if you're up for an experimental casual virtual hangout about WebAssembly, we can give it a go? :)

**Comment by aaronpowell** [comment]
Sure thing @ongmin. I'm in Sydney so I'm 3 hours ahead, so let's work out how to effectively set it up for everyone

**Comment by ongmin** [comment]
@aaronpowell just wondering first, does Wednesday 19th or Thursday 20th Feb evening, at 7pm (Singapore time) 10pm (Sydney time) work for you?

I will have to check which set up can support the most smoothly (guessing 10 people), and live stream the rest.

**Comment by aaronpowell** [comment]
Either date is fine for me @ongmin 😊

**Comment by ongmin** [comment]
I’ve posted the experimental meet-up link here: http://meetu.ps/e/HLtS9/10jMR/a

Let’s see how this short notice RSVP goes? I kept it for Wednesday 7pm. :)

Thanks for your flexibility! @aaronpowell

**Comment by ongmin** [comment]
All set thanks to Engineers.SG!

Zoom Meeting: https://zoom.us/j/168867062
Meeting ID: 168 867 062

Mike will also be live streaming on YouTube for us.

**Comment by aaronpowell** [comment]
I'll have a nap beforehand so I can stay awake 🤣

**Comment by ongmin** [comment]
Sorry about that but I hope it will be an amazing nap!! 😅

**Comment by aaronpowell** [comment]
Thanks for having me last night, it was a good bit of fun!

If anyone wants to get in contact the best way is twitter, I'm [slace](https://twitter.com/slace) on there, or you can follow my blog, https://www.aaron-powell.com.

The main link I had for folks to learn more is http://aka.ms/learn-wasm which is the start of a 6-part series on learning WASM and how it fits in a web application. The demo application I showed can be found here: https://github.com/aaronpowell/oz-dev-events and here's how I implemented search on my blog with Blazor [part 1](https://www.aaron-powell.com/posts/2019-11-29-implementing-search-in-blazor-webassembly-with-lucenenet/), [part 2](https://www.aaron-powell.com/posts/2019-12-11-optimising-our-blazor-search-app/).

**Comment by wlwl2** [comment]
Hi all, I changed my talk title. @ongmin I assume that the meetup will be a virtual one from my home?

**Comment by thchia** [comment]
@ongmin I think I overlooked that this is happening tomorrow... and I am not ready :( Can I be removed from the list of speakers? I’m so sorry 😨

**Comment by ongmin** [comment]
Hi @thchia @wlwl2 @ankita-stripe,

Stripe isn't able to host this month, sorry I haven't updated the news here. 
Shall we stick to a video cast for April instead? @thchia does April work for you? :)

**Comment by wlwl2** [comment]
@ongmin My data-viz talk has already been completed and prepared to be delivered as a fully-online talk or a regular talk. I want to delay my Time Converter and Joy-Con talk until long after this ordeal is over (at a more happier time). A video cast for April is fine.

**Comment by ankita-stripe** [comment]
I'm good with video cast for April too

**Comment by ongmin** [comment]
So let’s take a covid-19 break this month and gather again in April. :)

Thanks @wlwl2 for understanding and having the talk readied. Thanks @ankita-stripe too.

Stay safe and healthy!
